### ISSUE TITLE
Server mode crashes on re-analysis

### ISSUE THREAD

**Issue by jerhard** [goblint/analyzer#940]
Goblint crashes in server mode on re-analysis of code with:
```
Fatal error: exception Failure("there can only be one definition and one declaration per global")
```

When running goblint in server mode, as outlined [here](https://github.com/goblint/analyzer/issues/678#issuecomment-1335236833) e.g. by:
```
goblint --enable server.enabled --conf goblint.json --enable server.reparse file.c
```
and the submitting commands like:
```
{"id":0,"jsonrpc":"2.0","result":{"status":["Success"]}}
```
The first analysis that is triggered succeeds, while the subsequent one crashes with the Failure above. This happens even when the changes made to the file only affect white-space.

**Comment by jerhard** [comment]
For this program, the crash can be avoided when allowing multiple declarations (but not not definitions) in `addGlobal` in `CompareCIL.compareCilFiles` for a global. This is strange, that in server mode suddenly multiple declarations for a global appear.  Maybe `Cil` keeps multiple declarations when run by Goblint in server mode?

**Comment by jerhard** [comment]
Since we are only interested in the fact whether globals exist and what the varinfo is, I think it is ok to only keep a `varinfo option`. For all the declarations of global in a `Cil.file`, all the referenced `varinfo`s should be the same memory object.

**Comment by jerhard** [comment]
Also see [here](https://github.com/goblint/analyzer/issues/678#issuecomment-1336451451):
> > Might be useful to output which declaration it finds multiple of.
> 
> There are two functions for which multiple declarations are found in the incremental server run:
> 
> > static inline int pid_alive(const struct task_struct *p);
> > static __always_inline void data_access_exceeds_word_size(void)
> 
> Where `__always_inline` is defined as `inline __attribute__((always_inline))`. It is interesting that both functions are _static inline_ functions. However, there are quite some static inline functions in the same source files that do not have this issue. When running the incremental analysis multiple times in server mode, the number of declarations of the two functions above (or any others) does not further increase, i.e. the number of declarations of these functions stays at two starting from the first incremental run.

@stilscher and I looked at this, and it turns out that when `merge-inlines` is deactivated, this crash does not occur.

**Comment by stilscher** [comment]
I created a minimal example for this issue:
```
static inline void f(void);

static inline void f(void)
{
}

int main(void)
{
    return 0;
}
```
When analyzed with the following

<details><summary>configuration</summary>

```
{
	"files": ["file.c"],
	"kernel": true,
	"pre": {
		"kernel-root":"headers",
		"cppflags":["-D GOBLINT_NO_PTHREAD_ONCE", "-U __weak"]
	}
}
```
</details>

it fails in the incremental run (with empty diff) due to two declarations for `f` (with server mode and without). The problem does not appear when f is just a `static` instead of a `static inline` function. So there seems to be an issue with merging declarations for static inline functions. I will look into this further.

**Comment by stilscher** [comment]
I found out that this issue is caused by something rather unrelated to my previous hunch. The command for starting the server mode listed in the issue description above also hands over the file to be analyzed (in contrast to what is described in https://github.com/goblint/bench/blob/8ff038ba93c6cdbdb0ed3d6f992083d14047c719/gobpie-demos/chrony/story.md). 
In Goblint files that are handed over as arguments are added to the `files` option of the configuration unconditionally, no matter whether the respective file is already specified in the list. So if the respective file was already set in the configuration option as in the configuration above, it will be twice in the list of files analyzed by Goblint. 

I do not understand yet, why this is actually problematic, but it can be solved by removing the file name either from the command-line arguments or the configuration option `files`. Also, it doesn't seem to be about merging, because simply parsing and merging two copies of the above minimal test case file (with just one main method) in CIL works just fine. Also, the problem already occurs in the from-scratch run (a CIL warning about the same name being used for two globals is shown), but only leads to failure in the incremental run (because of the additional check when comparing two versions).

So a couple of ideas to avoid this issue:
- only add file arguments from the command-line that are not specified yet
- or alternatively disallow specifying the same file as a command-line argument and in the configuration option at the same time
- document the correct use of the server mode (e.g. in our Read The Docs)
- turn some CIL warnings that often indicate a serious problem into errors (like duplicate globals, conflicting main methods)
